lhotse/recipes/timit.py

```diff
@@ -0,0 +1,280 @@
+            elif part == 'DEV':
+                file_name = splits_dir / 'dev_samples.txt'
+            else:
+                file_name = splits_dir / 'tst_samples.txt'
+            wav_files = []
+            with open(file_name, 'r') as f:
+                lines = f.readlines() 
+                for line in lines:
+                    items = line.strip().split(' ')
+                    wav = os.path.join(corpus_dir, items[-1])
+                    wav_files.append(wav)
+                print(f'{part} dataset manifest generation.')
+                recordings = []
+                supervisions = []
+
+                for wav_file in tqdm(wav_files):
+                    items = wav_file.strip().split('/')
+                    idx = items[-2] + '-' + items[-1][:-4]
+                    speaker = items[-2] 
+                    transcript_file = Path(wav_file).with_suffix('.PHN')
+                    if not Path(wav_file).is_file():
+                        logging.warning(f'No such file: {wav_file}')
+                        continue
+                    if not Path(transcript_file).is_file():
+                        logging.warning(f'No transcript: {transcript_file}')
+                        continue
+                    text = []
+                    with open(transcript_file, 'r') as f:
+                        lines = f.readlines()
+                        for line in lines:
+                            phone = line.rstrip('\n').split(' ')[-1]
+                            if num_phones != 60: phone = phones_dict[str(phone)]
+                            text.append(phone)
+
+                        text = ' '.join(text).replace('h#', 'sil')
+
+                    recording = Recording.from_file(path=wav_file, recording_id=idx)
+                    recordings.append(recording)
+                    segment = SupervisionSegment(
+                        id=idx,
+                        recording_id=idx,
+                        start=0.0,
+                        duration=recording.duration,
+                        channel=0,
+                        language='English',
+                        speaker=speaker,
+                        text=text.strip())
+
+                    supervisions.append(segment)
+
+                    recording_set = RecordingSet.from_recordings(recordings)
+                    supervision_set = SupervisionSet.from_segments(supervisions)
+                    validate_recordings_and_supervisions(recording_set, supervision_set)
+
+                    if output_dir is not None:
+                        supervision_set.to_json(output_dir / f'supervisions_{part}.json')
+                        recording_set.to_json(output_dir / f'recordings_{part}.json')
+
+                    manifests[part] = {
+                        'recordings': recording_set,
+                        'supervisions': supervision_set}
+
+    return manifests
+
+def get_phonemes(num_phones):
```

Comment:
Should this function also handle 60 phones?

Also please raise an exception if somebody passes a different number than 39 / 48 (and maybe 60 if it makes sense).

Reply:
Please document this function.